WooCommerce/Classes/POS/Models/PointOfSaleViewStateCoordinator.swift

```diff
@@ -0,0 +1,30 @@
+import Foundation
+
+@available(iOS 17.0, *)
+protocol PointOfSaleViewStateResettable {
+    /// Resets all view state to its default values.
+    /// This should be called when starting a new cart to ensure the UI returns to its initial state.
+    func reset()
+}
+
+/// Coordinates view-specific state for the Point of Sale interface.
+/// This coordinator manages UI state that needs to persist across the POS experience
+/// but should be reset when starting a new cart/order.
+@available(iOS 17.0, *)
+@Observable final class PointOfSaleViewStateCoordinator: PointOfSaleViewStateResettable {
+    /// The currently selected item list type, shown when building the cart.
+    /// When we reset, the non-searched products list is shown for the new cart.
+    var selectedItemListType: ItemListType = .products(search: false)
+
+    /// The current search term being used to filter items.
+    /// This is used when `selectedItemListType` is in search mode.
+    /// When we reset, the search term is cleared for the new cart.
+    var searchTerm: String = ""
+
+    /// Resets all view state to its default values.
+    /// This should be called when starting a new cart to ensure the UI returns to its initial state.
+    func reset() {
+        selectedItemListType = .products(search: false)
```

Comment:
I wish we could avoid such a coordinator, but I don't have a suggestion right now.

If we organized our view-state correctly, we could achieve one out of two goals, either:

- Structure the views and state observation in a way that _maintains_ view state when we switch between .building <-> organizing 
- Structure the views and state observation in a way that _always cleans_ view state when we switch between building <-> organizing

However, when we tap back, we want the state to be maintained.
When we finish the order, we want the state to be cleaned.

I think we cannot describe that solely by how we structure the state, at least not in a straightforward way.

Reply:
I came to the same conclusion.

As long as we don't start making random changes to view state outside of the view, I don't think it's too bad. We may be able to improve our separation of a few other things from the logic too, now there's this place to put them.